tests/Avalonia.IntegrationTests.Win32/Infrastructure/AppManager.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Avalonia.Themes.Fluent;
+using Avalonia.Threading;
+
+namespace Avalonia.IntegrationTests.Win32.Infrastructure;
+
+internal static class AppManager
+{
+    private static readonly Lazy<Task<Dispatcher>> s_initTask = new(CreateUIThread, LazyThreadSafetyMode.ExecutionAndPublication);
+    private static readonly CancellationTokenSource s_cancellation = new();
+
+    public static void Stop()
+        => s_cancellation.Cancel();
+
+    private static Task<Dispatcher> CreateUIThread()
+    {
+        var tcs = new TaskCompletionSource<Dispatcher>();
+
+        var uiThread = new Thread(() =>
+        {
+            var appBuilder = AppBuilder
+                .Configure<Application>()
+                .UseWin32()
+                .UseSkia()
+                .SetupWithoutStarting();
+
+            appBuilder.Instance!.Styles.Add(new FluentTheme());
+
+            // Ensure that Dispatcher.UIThread is initialized on this thread
+            var dispatcher = Dispatcher.UIThread;
+            dispatcher.VerifyAccess();
+            tcs.TrySetResult(dispatcher);
+
+            dispatcher.MainLoop(s_cancellation.Token);
+        })
+        {
+            Name = "UI Thread"
```

Comment:
IsBackground = true? It doesn't seem to be explicitly killed either, so it should be at least background

Reply:
Just noticed cancellation token. That should be enough

Reply:
Indeed, if you remove the token and the runner doesn't stop. I left it as a foreground thread to mimic a real app as much as possible.